.github/workflows/gradle-latest-versions.yml

```diff
@@ -0,0 +1,45 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow.json
+name: Gradle latest versions tests
+
+on:
+  schedule:
+    - cron: "0 7 * * MON"
+  push:
+    branches:
+      - "gradleLatestVersionsTestManualTrigger"
+      - "szpak/gradleLatestVersionsGHActions"
+
+jobs:
+  gradle-compatTest:
+    name: "Compatibility tests with latest Gradle versions"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          java-version: 11
+          distribution: 'zulu'
+
+      - name: Fetch latest available Gradle versions
+        run: |
+          GRADLE_VERSIONS_TYPES_TO_FETCH="current release-candidate nightly release-nightly"
+          GRADLE_VERSIONS_OUTPUT_FILE=latest-gradle-versions.txt
+
+          for gv in $GRADLE_VERSIONS_TYPES_TO_FETCH; do
+            echo "Fetching latest Gradle version for '$gv'"
+            curl --silent --show-error "https://services.gradle.org/versions/$gv" | jq -r '.version | select( . != null )' >> latest-gradle-versions.txt
+          done
+
+          echo -e "\nGradle versions configured for compatibility tests (in $GRADLE_VERSIONS_OUTPUT_FILE):"
+          cat $GRADLE_VERSIONS_OUTPUT_FILE
+
+          echo "GRADLE_VERSIONS_OUTPUT_FILE=$GRADLE_VERSIONS_OUTPUT_FILE" >> $GITHUB_ENV
```

Comment:
Here it is mostly meaningless, but in the context of (potentially) externalize it to a separate shared action, I wonder, if setting the versions itself (with some "safe" separator) woudn't be a better option?